### PR TITLE
Gutenberg: Trigger E2E suite for v9.9.x upgrade

### PR DESCRIPTION
**Pull request by WunderBart**
Tracking issue: https://github.com/Automattic/wp-calypso/issues/49534.

Includes fix for failing tests where inserted block focus is stolen by the restore focus hook. See https://github.com/WordPress/gutenberg/issues/28932 for more details. 

#### ToDo:
A [fix for the focus issue](https://github.com/WordPress/gutenberg/issues/28932 ) mentioned above has been already merged and will be included in the v10.0 release. This PR might need to be reverted when we upgrade to that version.

#### What about failing specs?
See the summary here: https://github.com/Automattic/wp-calypso/pull/49599#issuecomment-778196819

### CODE REVIEW

**Comment by sirreal** [comment]
Rebased and running with 9.9.0 stable on edge.

**Comment by WunderBart** [comment]
Rebased and running with 9.9.1 stable on edge.

**Comment by fullofcaffeine** [comment]
Rebased and running with 9.9.2 stable on edge.

File: test/e2e/specs/wp-calypso-gutenberg-post-editor-spec.js [hunk 1]
```diff
@@ -73,7 +73,13 @@ describe( `[${ host }] Calypso Gutenberg Editor: Posts (${ screenSize })`, funct
 			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
 			await gEditorComponent.enterTitle( blogPostTitle );
 			await gEditorComponent.enterText( blogPostQuote );
-			await gEditorComponent.addImage( fileDetails );
+			const imageBlockId = await gEditorComponent.addImage( fileDetails );
+
+			// Force-select the image block. Workaround for  https://github.com/WordPress/gutenberg/issues/28932.
+			// This is needed or or the sidebar won't have the contextual settings for the block, and the step will fail.
```
Comment (fullofcaffeine) [inline test/e2e/specs/wp-calypso-gutenberg-post-editor-spec.js]: `or or` Oopsie

File: test/e2e/lib/gutenberg/gutenberg-editor-component.js [hunk 2]
```diff
@@ -394,11 +394,11 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
...
-		const insertedBlockSelector = By.css(
-			`.block-editor-block-list__block.${
-				initsWithChildFocus ? 'has-child-selected' : 'is-selected'
-			}[aria-label*='${ ariaLabel }']`
-		);
+		const insertedBlockSelectorStr = `.block-editor-block-list__block.${
+			initsWithChildFocus ? 'has-child-selected' : 'is-selected'
+		}[aria-label*='${ ariaLabel }']`;
+
+		let insertedBlockSelector = By.css( insertedBlockSelectorStr );
```
Comment (WunderBart) [inline test/e2e/lib/gutenberg/gutenberg-editor-component.js:401]: One of the things I wanted to refactor at some point - the `By` interface has "locator strategies" that return `locator`s, not `selector`s. A `selector` in this case is actually the string passed to the locator strategy (`css()` in this case). We use the selector naming for locators everywhere, so we should keep doing that to stay consistent, IMO. Just thought I'd mention the "locators" for future reference :) 

https://www.selenium.dev/documentation/en/getting_started_with_webdriver/locating_elements/

**Comment by WunderBart** [comment]
Because of [this PR](https://github.com/WordPress/gutenberg/pull/28191) that just got merged (planned for v10.1), this might end up not being a workaround at all:
> The first thing is that if you insert a block using the inserter, it's not "focused" anymore, which allows the inserter to stay open. From a screen reader perspective, this is correct. Moving the focus was the thing forcing us to actually close the inserter.

We need to keep changes from that PR in mind as they can cause more failed specs in our E2E GB suites.

_edit_

Looks like we have a fix already! :D https://github.com/WordPress/gutenberg/pull/28962

**Comment by WunderBart** [comment]
### Summary of failing [mobile specs](https://circleci.com/gh/Automattic/wp-e2e-tests-for-branches/140599):

✅ Currently, there are no failing `mobile` specs that run on Gutenberg edge sites. See details below.

#### Mailosaur-related (ignore, see p1612456065018600-slack-C1A1EKDGQ for context):
- `[WPCOM] Launch (mobile) @signup @parallel Launch a free site "before all" hook: Create a site as a new user for "Can launch a site"`
- `[WPCOM] Launch (mobile) @signup @parallel Launch when having multiple sites "before all" hook: Create 2 free sites as a new user for "Can start launch flow and abandon"`
- `[WPCOM] Launch (mobile) @signup @parallel Launch a free site "before all" hook: Create a site as a new user for "Can launch a site"`
- `[WPCOM] Launch (mobile) @signup @parallel Launch when having multiple sites "before all" hook: Create 2 free sites as a new user for "Can start launch flow and abandon"`

#### Non-Gutenberg-edge-related:
- `[WPCOM] Themes: All sites (mobile) Activate a theme @parallel when a theme more button is clicked when Activate is clicked Successful activation dialog should show the successful activation dialog`
  - Doesn't seem to be flaky, failing on master as well.
- `[WPCOM] Sign Up (mobile, en) Sign up for a non-blog site on a premium paid plan through main flow using a coupon @signup @visdiff Can Remove Coupon`
  - Flaky, passed locally.
- `[WPCOM] Plans: (mobile) Viewing a specific plan with coupon: @parallel @jetpack Remove any existing coupon`
  - Flaky, passed locally.

#### Gutenberg-related:
- `[WPCOM] Calypso Gutenberg Editor: Posts (mobile) Insert embeds: @parallel Can publish and view content`
  - Flaky, passed locally.


### Summary of failing [desktop specs](https://circleci.com/gh/Automattic/wp-e2e-tests-for-branches/140600):

✅ Currently, there are no failing `desktop` specs that run on Gutenberg edge sites. See details below.

#### Mailosaur-related (ignore, see p1612456065018600-slack-C1A1EKDGQ for context):
- `[WPCOM] Launch (desktop) @signup @parallel Launch a free site "before all" hook: Create a site as a new user for "Can launch a site"`
- `[WPCOM] Launch (desktop) @signup @parallel Launch a free site "before all" hook: Create a site as a new user for "Can launch a site"`
- `[WPCOM] Launch (desktop) @signup @parallel Launch when having multiple sites "before all" hook: Create 2 free sites as a new user for "Can start launch flow and abandon"`
- `[WPCOM] Launch (desktop) @signup @parallel Launch when having multiple sites "before all" hook: Create 2 free sites as a new user for "Can start launch flow and abandon"`
- `[WPCOM] Launch (desktop) @signup @parallel Launch a free site "before all" hook: Create a site as a new user for "Can launch a site"`
- `[WPCOM] Launch (desktop) @signup @parallel Launch a free site "before all" hook: Create a site as a new user for "Can launch a site"`
- `[WPCOM] Launch (desktop) @signup @parallel Launch when having multiple sites "before all" hook: Create 2 free sites as a new user for "Can start launch flow and abandon"`
- `[WPCOM] Launch (desktop) @signup @parallel Launch when having multiple sites "before all" hook: Create 2 free sites as a new user for "Can start launch flow and abandon"`

#### Non-Gutenberg-edge-related:
- `[WPCOM] Themes: All sites (desktop) Preview a theme @parallel when a theme more button is clicked when "Try & Customize" is clicked click try and customize popover`
  - Doesn't seem to be flaky, failing on master as well.
- `[WPCOM] Themes: All sites (desktop) Activate a theme @parallel when a theme more button is clicked when Activate is clicked Successful activation dialog should show the successful activation dialog`
  - Doesn't seem to be flaky, failing on master as well.
- `[WPCOM] Sign Up (desktop, en) Sign up for a non-blog site on a premium paid plan through main flow using a coupon @signup @visdiff Can Remove Coupon`
  - Flaky, passed locally.
- `[WPCOM] Plans: (desktop) Viewing a specific plan with coupon: @parallel @jetpack Remove any existing coupon`
  - Flaky, passed locally.

---

Overall, I don't see any specs that are related to Gutenberg v9.9. @fullofcaffeine's fix is working nicely, so I think we can ship this one and move on with the upgrade. cc @griffbrad

**Pull request merged**